File: src/main/java/com/pubnub/api/models/consumer/pubsub/PNMessageResult.java
package com.pubnub.api.models.consumer.pubsub;

import com.google.gson.JsonElement;

import com.pubnub.api.MessageType;
import com.pubnub.api.SpaceId;
import lombok.Getter;
import lombok.ToString;

@Getter
@ToString(callSuper = true)
public class PNMessageResult extends MessageResult {
    public PNMessageResult(BasePubSubResult basePubSubResult, JsonElement message, MessageType messageType, SpaceId spaceId) {
        super(basePubSubResult, message, messageType, spaceId);
    }
}

File: kotlin-publish/src/main/java/com/pubnub/publish/dependencies/models/PNMessageResult.kt
package com.pubnub.publish.dependencies.models

import com.google.gson.JsonElement

/**
 * Wrapper around an actual message received in [SubscribeCallback.message].
 */
data class PNMessageResult internal constructor(
    private val basePubSubResult: PubSubResult,
    override val message: JsonElement
) : MessageResult, PubSubResult by basePubSubResult
